model/transaction.go

```diff
@@ -111,7 +112,13 @@ func (e *Transaction) fields() common.MapStr {
 func (e *Transaction) Transform(_ context.Context, _ *transform.Config) []beat.Event {
 	transactionTransformations.Inc()
 
+	// Transactions are stored in a "traces" data stream along with spans.
+	dataset := datastreams.NormalizeServiceName(e.Metadata.Service.Name)
```

Comment:
The assumption that we don't need to differenciate spans from transactions is strong, but what if we are wrong? Isn't cheap and harmless to add an `apm.span` or `apm.transaction` dataset prefix, just in case?

Reply:
> Isn't cheap and harmless to add an apm.span or apm.transaction dataset prefix, just in case

It's not necessarily harmless, as more datasets will lead to more shards. We can always revisit this in the future.

Reply:
So how much it would hurt?
I guess its fine, but a similar argument can be done the other way around: by separating spans and transactions in different indices, it is likely that some queries in the APM UI can be changed to faster alternatives. 
We would not get this benefit if we revisit this decision in the future because the UI would have to query for both index name and index content... 
The opposite is not true: if we go with separated spans and transactions to start with, and later we find they don't add value (performance or user-flexibility wise), we would be able to optimize for less shards without any drawbacks.

Reply:
> So how much it would hurt?

We won't know for sure until we test.

This was discussed in the proposal, and the discussion concluded with this comment (from me):

"It seems to me that there's no clear answer here. In order to make progress I propose that we start out conservatively w.r.t. sharding, and have a combined data stream as described.
This will go out as experimental to start with, and we can adjust as necessary. We can replay some real data through the options to compare performance."

> I guess its fine, but a similar argument can be done the other way around: by separating spans and transactions in different indices, it is likely that some queries in the APM UI can be changed to faster alternatives.

Span docs are currently used in two places in the UI, AFAIK:
 1. displaying an individual trace waterfall
 2. walking traces for the service map

I'm confident (1) is going to be cheap either way. We know (2) is a sub-optimal approach to building the service map, and we're investigation options to move away from it.

> We would not get this benefit if we revisit this decision in the future because the UI would have to query for both index name and index content...

Not sure I follow. We don't query spans by themselves; only ever spans and transactions together. If we did want to query only spans in the future, then we could split the data sets and make `processor.event` a `constant_keyword`; that would mean only the spans data streams would be considered in the query. This can be done without changing the index pattern, and without breaking backwards compatibility.

Anyway, as mentioned above: let's go with this for now and test it at scale. We can use an ingest node processor to split the transactions and spans into separate streams for testing.